### ISSUE TITLE
Feature request: Allow tests to fail on ajax errors in a generic way

### ISSUE THREAD

**Issue by stefk** [DevExpress/testcafe#3167]
### What is your Test Scenario?

I'm testing an SPA which does a lot of ajax requests to an API in the background. While writing tests, I'd like to keep the focus on high-level scenari (i.e. testing from the point of view of the user, navigating, filling forms and so on) but still be informed of any ajax error that might happen in the background.

### What are you suggesting?

It would be really useful to be able implement a hook that would catch any http error and make the current test fail as soon as the error is caught, displaying an appropriate log.

### What alternatives have you considered?

Both `RequestLogger` and `RequestHook` offer a partial solution to this problem, however, as far as I know, it isn't currently possible to setup something that will:

- be attached to a whole fixture, and not every single test
- make the test fail as soon as the error is caught
- display useful information about the errors in a generic way

The closest I've came to this feature set is the following:

```js
import { RequestHook } from "testcafe";

class CustomRequestHook extends RequestHook {
  constructor() {
    super(null, {
      includeHeaders: true,
      includeBody: true
    });
    this.requests = {};
    this.failures = [];
  }

  onRequest(event) {
    this.requests[event._requestInfo.requestId] = event._requestInfo;
  }

  onResponse(event) {
    if (event.statusCode >= 400) {
      this.failures.push({
        url: this.requests[event.requestId].url,
        statusCode: event.statusCode,
        body: event.body.toString('utf8')
      });
    }
  }
}

const customHook = new CustomRequestHook();

fixture("Some fixture")
  .page("https://somepage.com")
  .requestHooks(customHook)
  .afterEach(async t => {
    if (customHook.failures.length > 0) {
      const failure = customHook.failures[0];
      await t.expect(failure.statusCode).lt(
        400,
        `Request to ${failure.url} failed with code ${failure.statusCode}. Response body was \n${failure.body}`
      );
    }
  });
```

As you can see though, this code has several problems:

1. It makes use use of the `_requestInfo` property, which is not part of the public API of Testcafe.
2. It has to define both a custom request hook and a `afterEach`.
3. It doesn't make the test fail when the error happens, but only after the body of the test has been executed.

Thanks in advance for your feedback!

**Comment by LavrovArtem** [comment]
Thank you for sharing your idea with us. We will discuss it and consider implementing it in one of our future product updates.

**Comment by yidea** [comment]
Any status on this, would be great if Testcafe can also provide the payload -on-fail option, just like the screenshot on fail

**Comment by arubtsov** [comment]
Hello @yidea,

We cannot provide you with any time estimates on this at the moment. Could you please describe your request in greater detail? What information do you need and how you need it to be stored or shown?
